File: Sorting/SelectionSort.py
def SelectionSort(a):
    l=len(a)
    max=0
    k=0
    count=0
    for i in range(l):
        max=a[0]
        k=0
        for j in range(1,l-i):
            if(max<a[j]):
                k=j
                max=a[j]
            count+=1
        a[l-i-1],a[k]=max,a[l-i-1]
    print(f"\n Selection sort takes {count} counts")
    return a

Arr=[56,55,89,42,46,32,28,25,21,10,98,56,63]
print("\n Before Sorting \n" + str(Arr))
Arr=SelectionSort(Arr)
print("\n After Sorting \n"+ str(Arr))

File: Sorting/SelectionSort.c
#include<stdio.h>

void SelectionSort(int a[], int len)
{
    int max=0,k=0,temp=0;
    for(int i=0;i<len;i++)
    {
        max=a[0];
        k=0;
        for(int j=1;j<len-i;j++)
        {
            if(max<a[j])          //If the max is less then we need to swap
            {
                max=a[j];
                k=j;
            }
        }
       a[k]=a[len-i-1];
       a[len-i-1]=max;
    }

}

void PrintArray(int a[],int len)
{
    printf("\n Array={");
    for(int i=0;i<len;i++)
    {
        printf("%d",a[i]);
        if(i+1<len)
        {
            printf(",");
        }
        
    }
    printf("}");
}

void main()
{
    int Arr[]={32, 12, 45, 10, 8, 92, 56, 74, 25, 66, 72, 77, 89, 40, 35},len=0;
    len=(sizeof(Arr)/sizeof(Arr[0]));
    printf("\n Before sorting");
    PrintArray(Arr,len);
    SelectionSort(Arr,len);
    printf("\n Afetr sorting");
    PrintArray(Arr,len);
}